pkg/provider/azure_storageaccount.go

```diff
@@ -462,7 +470,7 @@ func (az *Cloud) createVNetLink(ctx context.Context, vNetLinkName, vnetResourceG
 		Location: &location,
 		VirtualNetworkLinkProperties: &privatedns.VirtualNetworkLinkProperties{
 			VirtualNetwork:      &privatedns.SubResource{ID: &vnetID},
-			RegistrationEnabled: to.BoolPtr(true)},
+			RegistrationEnabled: to.BoolPtr(false)},
```

Comment:
why make this change?

Reply:
According to the [document](https://learn.microsoft.com/en-us/azure/dns/private-dns-virtual-network-links#registration-virtual-network), every virtual network can only have one registration zone associated with it. Since we need to create two   virtual network link, we cannot set `RegistrationEnabled` as true for both of them. 

And auto-registration is used for virtual machine: `A DNS record gets automatically created for any virtual machines you deploy in the virtual network.` So no need to set it as true.